src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -606,14 +610,6 @@ impl fmt::Display for LogicalAgg {
 
 impl ColPrunable for LogicalAgg {
     fn prune_col(&self, required_cols: &[usize]) -> PlanRef {
-        let upstream_required_cols = {
-            let mapping = self.o2i_col_mapping();
-            FixedBitSet::from_iter(
-                required_cols
-                    .iter()
-                    .filter_map(|&output_idx| mapping.try_map(output_idx)),
-            )
-        };
```

Comment:
It is unnecessary to union `upstream_required_cols`, as it is always a subset of `group_key_required_cols` below.